Add security team charter.

tim-schilling opened the pull request:
Hi all (@django/django-security-team, @django/steering-council), here's a new draft of the Security Team charter that's available to review. There are a few changes highlighted to help us all be more effective and to help the team build for the future. I'd like to hear your opinions and concerns on what you think would work well and what won't.

The reason we're doing this with all the teams is to help the Fellows operate a bit more smoothly in our community. For other teams, it's defining a standard way to communicate with them and their expectations. That applies to the security team, but we'd like to explore dedicating people to a specific role too (see Report Triagers).

Proposed changes from existing operations:
- Creating report triager role for initial triaging of reports
- Generating an annual report
- Allowing people to self-nominate to join the security team
- Annual opt-in for members
- Creating chair and co-chair roles

### Creating report triager role
The goal here is to alleviate some of the daily workload from the Fellows. The Fellows are involved in many areas of the code base on a daily basis. They have a high awareness of what's going on and can help facilitate changes much more effectively than those who contribute less frequently.

The initial triaging of security reports is something that must be done which is why the Fellows are doing it. Helping maintain our security standards is one of their directives.

However, the initial review of a security report to determine if it's likely a valid report, what components it touches, and determining who likely will want to be involved doesn't require a Fellow. It requires legwork for someone to do the communication between the relevant parties.

Keep in mind, this change isn't necessarily asking an existing member to fill this role. If someone wants to be more active on the communication front, that's great. If not, we can recruit new team members and use this to set expectations for what their involvement would focus on.

### Generating an annual report
The purpose here is to help communicate with the broader community about your work and identify if other help is needed. It's possible there are security interested folks out there that would like to do the legwork to build out tooling to help the security team or security focused parts of the community. By bringing attention to it, we may be able to motivate newer contributors.

### Allowing members to self-nominate
There are a few reasons why this is being suggested:
- People who may be qualified who aren't in the personal network of the team, currently aren't eligible to join
- Some people may continue to be on this team because they think there's nobody to replace them
- Having a lower barrier to entry to join should lead to more new members, fresh ideas and energy to innovate
We can also change this again in the future if we find it's unhelpful or noisy. This doesn't need to be permanent, but it seems like something helpful to be explored.

### Creating a chair and co-chair role
With the addition of responsibilities that are ideally outside the Fellows purview, we'd want to identify one or two people who will help facilitate these actions. They also exist as the contact points for other teams.

## Other questions
- Does the team want to revise how to remove a member from the team? It currently states it requires full consensus from the other members
- Do you want to revisit the scope of responsibilities?

Thank you to @jefftriplett and @thibaudcolas for helping with the draft.

tim-schilling committed:
Add draft of security team definition.

Co-authored-by: Thibaud Colas <thibaudcolas@gmail.com>

ubernostrum commented on active/security.md:
```diff
@@ -0,0 +1,93 @@
+
+### How to join
+
+Any person can volunteer to join the security team by submitting a Google Form (TODO: Create link). The team/WG will vote (50%+1) to approve/deny new members; the team/WG will directly vote on new Chair/Co-Chairs.
```
I do not think the security team should have a standing application form. If new members are added only when needed, and the need does not arise very often, a continually-open application form has multiple problems:

* Creates an expectation from applicants that new members are being accepted all the time, when that's not the case
* Even with appropriate messaging about whether new members are currently being accepted, creates the likelihood of applications waiting for extremely long periods (potentially years), and needing to re-check whether the applicant is still interested/applications are still current/etc. if and when a new member acceptance period does finally open up.

So this should instead define that the team will, when the team decides, open up applications for new members, but that at other times there will be no standing application/volunteer form.

ubernostrum replied:
The thing with a security team is that "larger" is not automatically "better", because more people means more difficulty coordinating and more difficulty managing access to sensitive information. And people who work in security tend to be aware of that, so will expect that the Django security team is likely to be kept on the small side and recruit when needed rather than be constantly taking in new members (in fact, I'd expect an always-open process to bring more people in to the security team would actually be seen as a red flag by a lot of people who do security professionally).

Is there some sort of specific target size that the SC/DSF Board have in mind for this team that's not coming through in this discussion? Is there some other reason why, despite getting negative feedback multiple times, this application/recruiting process keeps getting pushed in these drafts?

tim-schilling replied:
I'm going to revise this to be invite only to match the language in the Ops team charter to align to the current processes. The team can adjust this in the future.

tim-schilling replied:
Revised in https://github.com/django/dsf-working-groups/pull/56/commits/183e6ebb35535a9588600dcbf7fb05885609dd0d. I'm resolving this thread.

jacobtylerwalls commented on active/security.md:
```diff
@@ -0,0 +1,93 @@
+The team has discussions in two places:
+
+1. Formal and sensitive discussions on the mailing list: security@djangoproject.com
+2. Informal and team discussions on the DSF Slack in the private channel `#security-team`
```
Just wondering: does this already exist, or is this proposed?

ubernostrum replied:
Is there even a need for a Slack channel for the security team? For the things the team handles, the fewer places that contain the information, the better.

RealOrangeOne replied:
How does one join the DSF Slack? :eyes: It was only recently I learned it existed. I don't think it's part of the team onboarding process.

tim-schilling replied:
```suggestion
```

shaib replied:
The slack channel is active, so @tim-schilling 's suggestion in this thread can be dropped.

tim-schilling replied:
Marking this resolved too. Thanks for pointing that out @shaib

MarkusH commented on active/security.md:
```diff
@@ -0,0 +1,93 @@
+### Role definitions
+
+- Chair / Co-Chair: Responsible for coordinating the group, scheduling meetings, renewing the group’s membership, and ensuring that the group’s activities align with its scope and responsibilities.
+- Report triagers: These team members are responsible for acknowleding and triaging reports initially to determine likelyhood of security concern and severity.
```
Do we need a dedicated role for this? Works without that since ... _ever_??

tim-schilling replied:
I'm marking this as resolved due to the email thread and agreement on us needing to have a group of people identified who are doing more triage work. The details of how this is formalized in the charter is still being worked out.

tim-schilling commented on active/security.md:
```diff
@@ -0,0 +1,93 @@
+  - Shai Berger
+  - Simon Charette
+ 
+Note: The DSF Board President has access to the security mailing list, but does not otherwise participate in the team’s activities. This is mentioned for the sake of transparency.
```
The DSF president is the Google org admin which means they have access to everything. That includes the security mailing list.

RealOrangeOne replied:
Ah, I misinterpreted "has access to" as "receives emails" :+1:

RealOrangeOne replied:
As someone who admins a few fairly large Google Workspace organisations, I'm happy to have a poke at resolving this, rather than documenting it.

shaib replied:
So, the issue remains open: Regardless of whether it was by design or not -- do we still want it?
(personally -- as the president is not a member of @django/django-security-team nor the security-team's Slack channel, I fail to see why they should be a member of the mailing list).

tim-schilling replied:
@shaib I think this may be better addressed in the DSF Slack with the Board. As I understand it, this line reflects how things have been operating for a while.

jefftriplett replied:
While I mostly tune out unless pinged, the historical context is a mix of legal responsibility and governance oversight.

From a legal standpoint, the DSF President is an officer of the foundation. If there were ever a security incident serious enough to create legal exposure for Django or the DSF, the President would likely be involved in working with counsel and helping represent the organization.

From a governance standpoint, that access has also served as a simple canary in the coal mine, indicating that the lights are still on. The President is not part of the day-to-day security process, but can notice if reports go unanswered for an unusually long time and help ensure the right people are aware.

And to confirm, yes, the DSF President has had this access since at least Russell’s time as President, and it may predate him as well. It has been a longstanding practice rather than something newly introduced.

So the access is less about participation and more about accountability, continuity, and transparency.

nessita replied:
Thank you @jefftriplett for the clarifying comment. I think it would be a good idea to capture this valuable information in the charter, how about something like:
```suggestion
Note: The DSF Board President has access to the security mailing list for governance oversight and legal accountability purposes. This is a longstanding practice and is mentioned for the sake of transparency.
```

shaib replied:
Thanks @jefftriplett , that helps understand a lot. But still, the current wording, and even @nessita 's suggestion, don't explain the discrepancy between the mailing list and other team communication channels. I'm not sure it can be done very concisely, and since this is just a side note, it shouldn't get out of hand.

I tried to come up with something that's still short, and more concrete than "governance oversight and legal accountability", and failed. So I'm +0.5 on the suggestion.

jefftriplett replied:
We recently adjusted the Fellow's contractual language to read... "designated reporting channels and ensure they are acknowledged and handled promptly" on the off chance that it helps to reword it. 

Maybe this? 

> Note: The DSF Board President has access to the designated reporting channels for governance oversight and legal accountability. This is a longstanding practice, noted here for transparency.

That drops "mailing list," which is more of a designated reporting channel, vs. adopting Slack, Signal, or something else for internal communications. I think governance-wise, we just want to know that messages that are being reported/incoming are getting handled, and something is going out on the off chance that everyone quiet quits, and the rest we already outlined.

shaib replied:
Yes, thank you both. I like this a lot.

sarahboyce replied:
Resolved by 9c1f116769bae4746ed4b14e5fad49bb1e5b6c73

tim-schilling commented on active/security.md:
```diff
@@ -0,0 +1,93 @@
+### Role definitions
+
+- Chair / Co-Chair: Responsible for coordinating the group, scheduling meetings, renewing the group’s membership, and ensuring that the group’s activities align with its scope and responsibilities.
+- Report triagers: These team members are responsible for acknowleding and triaging reports initially to determine likelyhood of security concern and severity.
```
As many that helps the team be sustainable I imagine. How many would be needed if each person is spending 1 hour per week?

tim-schilling replied:
Marking as resolved due to our email discussion.

tim-schilling commented on active/security.md:
```diff
@@ -0,0 +1,93 @@
+
+The security team is responsible for [Django’s security policies](https://docs.djangoproject.com/en/dev/internals/security/). This includes:
+
+- Reviewing security reports via security@djangoproject.com
```
Ah, it sounds like we may need to flesh this section out a bit further then. I will need some help with that.

sarahboyce replied:
```suggestion
- Reviewing security reports
```
I personally feel we can remove platform specific information. We link the security policy docs which has information on reporting. This currently doesn't mention HackerOne but if we want to document this. we should document it there. I don't think we want to have to update this page if we chose to update our reporting process

nessita reviewed (commented):
Thank you so much @tim-schilling for putting this together. 🌟 

I see a lot of the Fellows’ asks from the last 12 months being sensibly laid out in this proposal.

I added my personal thoughts as well as the Fellows POV based on what we have been discussing on an ongoing basis for several months now. Happy to continue iterating!

tim-schilling commented on active/security.md:
```diff
@@ -0,0 +1,93 @@
+
+- Becoming disqualified by the Code of Conduct working group
+- A vote of the Steering Council
+- The full consensus of the rest of the Security Team
```
Correct

nessita replied:
I think this is too strong and we should remove it. May I ask where did you see this?

tim-schilling replied:
I think I pulled this from the accessibility team's requirements. https://github.com/django/dsf-working-groups/blob/main/active/accessibility.md?plain=1#L87

tim-schilling replied:
Yeah, I couldn't find what I thought was a security team document. It had to be the accessibility team charter since I was using that as inspiration for this one.

shaib replied:
~~I'm with @nessita~~ -- we don't need this; if a member of the security team is so troublesome that most (even not all) of the rest of the team want to remove them, it is better to require the involvement of the Steering Council.

shaib replied:
After writing the above, I saw that @nessita was actually suggesting only the removal of the "full consensus" requirement. I disagree.

tim-schilling replied:
@shaib If you're suggesting we remove this line entirely and only allow the CoC and SC to remove a member, I don't think you're in alignment with @nessita. She believes (and I agree) that a smaller group of members who agree to remove a person should be sufficient. In https://github.com/django/dsf-working-groups/pull/56/files#r3124348422 it outlines it as "A vote (50%+1) of the existing team members"

Edit: I wrote the above before I saw the edits and follow-up. Whoops! Sorry to reiterate your conclusion.

shaib replied:
Right.

tim-schilling commented on active/security.md:
```diff
@@ -0,0 +1,93 @@
+- Why do you want to join the team?
+- What is your history of using Django as a developer?
+- What is your history of contributing to Django?
+- What security experience do you bring that would be helpful to the team?
```
I feel like that may be covered in "What is your history of contributing to Django?"

shaib replied:
This can also be resolved, in view of changes in https://github.com/django/dsf-working-groups/commit/183e6ebb35535a9588600dcbf7fb05885609dd0d

tim-schilling committed:
Apply suggestions from code review

Co-authored-by: nessita <124304+nessita@users.noreply.github.com>

tim-schilling commented:
There was agreement [via email] amongst the team on the following aspects:
- An advisor-type set of expectations for all team members by default
- A explicit role of Responders/Triagers that will be more active on the legwork of security reports (see Natalia's "Tasks (listed, but not discussed)" from the Notes document)
- People can adopt/remove the Responders/Triagers role as they are available and interested
- Fellows both facilitate and pick up any Responders/Triagers slack on reports

I'll be working update the document today and then facilitating further discussion on the other open points.

tim-schilling committed:
Clean-ups on the charter.

tim-schilling committed:
Clean-ups on the charter.

tim-schilling committed:
Better define report triager role with tasks. Add basic participation requirement for all members.

tim-schilling committed:
Update SC liaison

tim-schilling committed:
Update the responsibilities of a triager.

jacobtylerwalls commented on active/security.md:
```diff
@@ -0,0 +1,116 @@
+- Applying, backporting and releasing those fixes as patches and Django releases
+- Communicating with reporters
+- Communicating with the public about security releases
+- Communicating with operating-system vendors and other distributors of Django
```
We are going to need a line item here for items relating to being a CVE Numbering Authority (CNA), such as maintaining our membership in good standing, deciding whether and how to cast votes in elections.

This will free up Thibaud and Sarah A from having to monitor the cna emails until we get this responsibility transferred. I can suggest some wording

```suggestion
- Communicating with operating-system vendors and other distributors of Django
- Maintaining the DSF's status as a CVE Numbering Authority (CNA) and casting votes (or abstaining) in CNA elections
```

tim-schilling committed:
Add responsibility of maintaining status as a CNA

Co-authored-by: Jacob Walls <jacobtylerwalls@gmail.com>

sarahboyce commented on active/security.md:
```diff
@@ -0,0 +1,117 @@
+
+- Reviewing security reports via security@djangoproject.com
+- Evaluating and developing fixes for confirmed security issues
+- Applying, backporting and releasing those fixes as patches and Django releases
```
Technically only a merger and releaser would be able to do this. Perhaps we need to state that at least one merger and releaser needs to be on the team and that this is covered by the Fellows

shaib replied:
Other docs describe it as "the Security Team asks for a release to be made" (DEP 19 inherits this from DEP 10/12, IIRC). That description seems more appropriate in the abstract, but the practice is as @sarahboyce says -- the fellows, being Mergers, Releasers and members of the Security Team, collect security fixes into releases without any actual "request" from the Team.
I'm not sure which of these the Charter should reflect, probably both somehow.

tim-schilling replied:
I'm going to drop "releasing", but the others seem reasonable and can be interpreted to not require the mergers permission.

sarahboyce commented on active/security.md:
```diff
@@ -0,0 +1,117 @@
+- Communicating with operating-system vendors and other distributors of Django
+- Maintaining the DSF's status as a CVE Numbering Authority (CNA) and casting votes (or abstaining) in CNA elections
+
+## Membership
```
Should we mention that Fellows are automatically members of the security team and therefore bypass any normal new membership process? Fellows will have been granted the releaser and merger roles and so can (at a minimum) handle the security releases

manfre replied:
It's worth mentioning.

shaib replied:
I think maybe "Fellows" should be a separate role in the list below.

tim-schilling replied:
Mentioned the Fellows in the membership section in https://github.com/django/dsf-working-groups/pull/56/commits/183e6ebb35535a9588600dcbf7fb05885609dd0d

sarahboyce commented on active/security.md:
```diff
@@ -0,0 +1,117 @@
+- **Chair / Co-Chair:** Responsible for coordinating the group, scheduling meetings, renewing the group’s membership, and ensuring that the group’s activities align with its scope and responsibilities. The Chair and Co-Chair roles should be re-evaluated annually by the team.
+- **Report Triagers:** Acknowledge and triage initial reports and communicate with reporters.
+
+#### Report Triager
```
We have
```
- Participate in the process for at least one security report a year
```
Should we have a minimum expectation target here as well so folks applying understand the work involved. For example:
```
- Triage 1 report every two weeks
- Join 1 team meeting a year
```
I think I would also want members to communicate holidays so we are aware that they are away.

manfre replied:
Joining at least 1 team meeting per year seems like a suitable check to identify those who are inactive.

I like the generic "Participate in the process..." more than "Triage 1 report".

shaib replied:
In this section, dealing specifically with Triagers; and with the definition of what "Traiging" means in the lines immediately following -- I prefer @sarahboyce 's wording.

shaib replied:
On the other hand, see below @nessita 's suggestion of having, instead of the Triager/Regular distinction, a Regular/Special-Expert distinction, where the Special Experts are expected to participate once a year, but Regular participation expectations are higher. FWIW, that is a much better reflection of the current team's composition (but also a reflection of how hard it is, currently, to get new members).

nessita replied:
Right, I would rather we have something like this, which feels more realistic and also more aligned with the needs of the Team:

- Triage at least one report a month
- Join at least one team meeting in a 6 months period

For the "Historical/Special-Expert" role, I think we could have:

- Participate in the process for at least one security report a year

sarahboyce replied:
Resolved by 9c1f116769bae4746ed4b14e5fad49bb1e5b6c73

sarahboyce commented on active/security.md:
```diff
@@ -0,0 +1,117 @@
+- What is your history of contributing to Django?
+- What security experience do you bring that would be helpful to the team?
+
+(TODO: Define cadence of reviewing applications)
```
I think "Applicants are reviewed as part of the team meetings"
I believe the meetings are monthly but I don't think we need to document that it is every month as discussions might take longer to conclude a decision

tim-schilling replied:
This will be removed as per https://github.com/django/dsf-working-groups/pull/56/files#r3110470338

sarahboyce reviewed (commented):
Thank you for the work that has been invested in this everybody ⭐ 

I have added a few comments to hopefully help progress. Generally happy with what is here already 👍

shaib commented on active/security.md:
```diff
@@ -0,0 +1,117 @@
+
+Every member can adopt and step back as a Report Triager as their schedule allows.
+
+The responsibilities of a Report Triager are as follows:
```
```suggestion
Triaging a report includes: 
```

tim-schilling replied:
Sorry, I disagree with this suggestion. I think the mention of responsibility is important for this part of the document.

shaib replied:
Point taken (again, I'm unable to resolve the thread myself)

shaib commented on active/security.md:
```diff
@@ -0,0 +1,93 @@
+
+The team has discussions in two places:
+
+1. Formal and sensitive discussions on the mailing list: security@djangoproject.com
```
I subscribe to the later suggestion above, that we shouldn't include our tools for accepting reports in the charter.

tim-schilling replied:
@shaib this is more describing how the team communicates with each other, less on how reporters communicate with the team. Teams and working groups that have an email list have listed them in this section of their charters.

shaib commented on active/security.md:
```diff
@@ -0,0 +1,117 @@
+
+1. Formal and sensitive discussions on the mailing list: security@djangoproject.com
+2. Informal and team discussions on the DSF Slack in the private channel `#security-team`
+
```
```suggestion
3. Monthly video-conference meetings

```

shaib reviewed (commented):
A few updates and other notes

tim-schilling committed:
Remove specification of where security reports are filed.

Co-authored-by: Sarah Boyce <42296566+sarahboyce@users.noreply.github.com>

tim-schilling commented on active/security.md:
```diff
@@ -0,0 +1,117 @@
+
+- Reviewing security reports
+- Evaluating and developing fixes for confirmed security issues
+- Applying, backporting and releasing those fixes as patches and Django releases
```
```suggestion
- Applying and backporting those fixes
```

tim-schilling committed:
Remove releasing as a responsibility as it implies they have the ability to actually release.

They can only request a release, which is similar effectively, but could be confusing.

tim-schilling committed:
Add video-conference meetings to communication mechanisms.

Co-authored-by: Shai Berger <shai@platonix.com>

tim-schilling committed:
Remove application process. Add Fellows as automatic members.

tim-schilling commented on active/security.md:
```diff
@@ -0,0 +1,93 @@
+ 
+## Future membership
+
+The team does not have a fixed size. The team decides when new members are needed. New members are chosen from a list of volunteers. If there are no qualified volunteers the team will place an advertisement on the Django website.
```
I've revised the language for this in https://github.com/django/dsf-working-groups/pull/56/commits/183e6ebb35535a9588600dcbf7fb05885609dd0d. I think this thread can be resolved now.

shaib replied:
(for some reason, I seem unable to resolve this myself)

tim-schilling committed:
Reword the comms text to make sense with third option.

nessita commented on active/security.md:
```diff
@@ -0,0 +1,93 @@
+ 
+### Role definitions
+
+- Chair / Co-Chair: Responsible for coordinating the group, scheduling meetings, renewing the group’s membership, and ensuring that the group’s activities align with its scope and responsibilities.
```
Since these comments, I have arranged monthly meetings happening on the second Wednesday of each month and so far we have held 2, with decent attendance, and in my opinion these have proven to be necessary and valued.

nessita replied:
(In other words, I think this can be resolved with the current wording.)

nessita commented on active/security.md:
```diff
@@ -0,0 +1,113 @@
+
+- Reviewing security reports
+- Evaluating and developing fixes for confirmed security issues
+- Applying and backporting those fixes
```
I know is a technicality, but members of the team can't apply fixes (can not merge into the repo), nor they can release artifacts. These actions are still done by mergers and releasers, so perhaps we should include some wording that at least one of each team should be a member, and that this requirements is most commonly fulfilled by the Fellows?

tim-schilling replied:
I'm happy to change the wording.

My opinion is that we should be against adding requirements about who should be on the team though. It's codified in DEP 19 (and 10) that the security team can direct a PR to be merged in. That should be sufficient.

As context, I originally interpreted this as more of helping prepare the fix and the commits to be backported. Not the actual merge and release mechanisms.

nessita replied:
That makes sense, thank you. I guess where I see some confusion is that for security fixes, in most cases, there are no PRs that could be instructed to be merged, thought we could say "patches" which is more realistic. In any case, I'm happy to leave it without much detail but I would value if a minimal clarification could be added so future readers are not confused like me.

tim-schilling replied:
```suggestion
- Applying and backporting those fixes in coordination with Mergers and Releasers
```
Is this the type of minimal clarification that would help?

sarahboyce replied:
Resolved by 9c1f116769bae4746ed4b14e5fad49bb1e5b6c73

nessita commented on active/security.md:
```diff
@@ -0,0 +1,113 @@
+- Communicating with reporters
+- Communicating with the public about security releases
+- Communicating with operating-system vendors and other distributors of Django
+- Maintaining the DSF's status as a CVE Numbering Authority (CNA) and casting votes (or abstaining) in CNA elections
```
This is great, but perhaps we should also include that we have the obligation to:
* handle CVE requests sent to cna@djangoproject.com
* issue CVEs for Django and projects under the Django org (we could link to the CNA scope perhaps)

tim-schilling replied:
Does the first one fall under "Reviewing security reports"? We don't need to have all the details here.

Agreed on the second. Is there a way we can capture all the CVE responsibilities in one statement? If not, we can break it up.

nessita replied:
For the first one, I wanted to signal those CVE requests that are not a security report but an actual CVE number request, like the one we could get from Carlton for a CVE for daphne.

EDIT: sorry I'm being confusing. I think leaving just the second item suffices.

tim-schilling replied:
```suggestion
- Maintaining the DSF's status as a CVE Numbering Authority (CNA) and casting votes (or abstaining) in CNA elections
- Issue CVEs for Django and projects under the Django organization
```

I think this is what you're referring to

sarahboyce replied:
Resolved by 9c1f116769bae4746ed4b14e5fad49bb1e5b6c73

nessita commented on active/security.md:
```diff
@@ -0,0 +1,113 @@
+
+## Membership
+
+- Chair:
```
I believe we settled on me being the chair for a period.

tim-schilling replied:
```suggestion
- Chair: Natalia Bidart
```

nessita commented on active/security.md:
```diff
@@ -0,0 +1,113 @@
+- Initial assessment
+- Request help from team/experts if necessary
+- Progress to resolution
+    - For valid reports, hand-off to team member to own the report
```
Could this be a tiny bit looser and allow for the same triager to progress it?

tim-schilling replied:
```suggestion
    - For valid reports, hand-off to a team member to own or retain yourself to progress
```

shaib replied:
One important consequence of this definition, which I think should be highlighted, is that each report is to have a more-or-less formal designated owner, at each point since its first acknowledgment. This is, as far as I'm aware, not how we do things now, and keeping it real requires a more formal process of managing reports (e.g. so long as reports are actually handled on the mailing list, it's very hard to ascertain who the owner of each report is, and if all reports are actually owned).

Note, I'm not at all opposed to this -- I'm sure it would make it much easier to verify that things are not falling between cracks -- but to be effective it also requires tooling, and while such tooling has been discussed on the team, this is very much a work-in-progress.

sarahboyce replied:
So, as this should be a living document, I think we should just say "Optionally engage in the progress to resolution" so that we explicitly say a triager can still help in the creation of fixes/reviewing of fixes for the reports they triaged. 
Processes around ownership we can add later once we actually have a proper process and tooling to support this (of course there is an informal process that the Fellows track things).

shaib replied:
I tend to agree -- but note that this takes all the teeth out of the "Responsibility" language (see https://github.com/django/dsf-working-groups/pull/56/changes#r3110503550 -- that's a resolved comment on Line 55 in the current version).

shaib replied:
Ownership/responsibility is still an open issue AFAICT

sarahboyce replied:
```suggestion
    - For valid reports, state reasoning/opinion and whether wish to engage further in the progress to resolution (e.g. create a PR to resolve the issue)
```
Perhaps something like this

sarahboyce replied:
I believe I have resolved this in 9c1f116769bae4746ed4b14e5fad49bb1e5b6c73
Please make a concrete suggestions if not

nessita commented on active/security.md:
```diff
@@ -0,0 +1,113 @@
+
+- Becoming disqualified by the Code of Conduct working group
+- A vote of the Steering Council
+- The full consensus of the rest of the Security Team
```
As part of our ongoing work to support the Security Team, we (the Fellows) have been thinking about how the future membership process could be improved and defined. We would like to propose the following, with the goal of keeping things practical and lightweight while making it easier to bring in new members:

```suggestion
If the team needs to make a call for volunteers, it will be posted on the [djangoproject.com blog](https://www.djangoproject.com/weblog/) and the [Django Forum](https://forum.djangoproject.com). Also, any DSF member or member of a DSF working group may submit a nomination, including a self-nomination, to the team's mailing list (security@djangoproject.com).

Nominations are discussed within the team, with a decision made at the next Security Team meeting, provided at least a two-week window has passed since the nomination was submitted. A nomination is approved when at least one existing Security Team member votes in favor and no member objects. Team members are encouraged to share their position via the mailing list before the meeting.

New members complete a 3-month onboarding period during which all external communications (with reporters, vendors, or distributors) must be reviewed by an existing member before sending. This ensures each report is handled consistently and no valid reports are inadvertently dismissed.

The membership will operate as follows:

- There is no upper limit to the team size
- All Django Fellows are automatically added to the Security Team
- A Django Fellow contract termination removes the Fellow from the Security Team
- Each year, every non-Fellow member will need to reaffirm their membership with the team
- A member can leave for any reason

Members can also be removed by:

- Becoming disqualified by the Code of Conduct working group
- A vote of the Steering Council
- A vote (50%+1) of the existing team members
```

cc/ @sarahboyce @jacobtylerwalls

shaib replied:
Just to make sure -- 

- the wording "**Also**, any DSF member or member of a DSF working group may submit a nomination, including a self-nomination" (emphasis mine) is intended to make nominations open even when we don't ask for them?
- The onboarding period is for all new members -- including new Fellows?
- Is a Fellow free to leave the Security Team while keeping the Fellow position?

As noted above, I disagree with the idea that the team should be able to kick out members without Steering Council approval.

sarahboyce replied:
> nominations open even when we don't ask for them?

Yes. For example, Jake Howard was recommended by Thibaud due to his Wagtail security involvement. There was no explicit call for applications. I thought Jake's onboarding and involvement has been very positive and I would like this spontaneous addition to be possible. 
If there is a point we think the team shouldn't have new members, we can have a standard response and reach out to those folks again if we change that decision.

> The onboarding period is for all new members -- including new Fellows?

Personally, yes (unless the new fellow was an existing team member of course). This does motivate me to say let's reduce the onboarding from 3 months to 2.

> Is a Fellow free to leave the Security Team while keeping the Fellow position?

Beautiful question. I would say no. Our current contracts have security as one of the main priorities of the role. You cannot fullfill the contract without being a member. If the contract/Fellow definition changes, maybe this will change.
But a Fellow could still be expelled by the criteria above (and as a consequence the Fellowship WG should decide whether to terminate the contract).

Does that make sense to you? Do you think any of those bits should be spelled out?

sarahboyce replied:
> As noted above, I disagree with the idea that the team should be able to kick out members without Steering Council approval.

Trying to think about this fresh over some coffee this morning.
I think looking at this from a different angle might help here. I see three main scenarios for removal:

- Inactivity: If a member is inactive per this document’s definition, the team should be able to remove them independently, with a note of thanks and an open invitation to return. Steering Council involvement doesn’t seem necessary here, nor does unanimous agreement from the team.
- Security or malicious concerns: The team should be able to temporarily suspend access if there is a credible concern, notifying the individual and pausing access during investigation. A final decision (e.g., permanent removal) should involve the Steering Council. Temporary action should be quick. I would propose this requires at least two team members voicing a concern.
- Code of Conduct or behavioral issues: These cases are often complex and an ugly/messy business, so I am comfortable with the Steering Council (and/or CoC team) being mediators who handle the investigation and make the final decision (perhaps with a vote).

What do we think?

LilyFirefly replied:
For the final point I wouldn't feel comfortable with the Steering Council holding responsibility for or leading anything Code of Conduct related. I think that should be the responsibility of the CoC team and they can request support from the Steering Council if desired.

sarahboyce replied:
I hadn't read through the new CoC guidelines in detail (such as https://github.com/django/code-of-conduct/blob/main/enforcement-ladder.md#django-code-of-conduct---enforcement-ladder). 
I think I agree point 3 should be managed by the CoC team rather than the Steering Council. I agree they have the expertise and authority in that area

jacobtylerwalls replied:
I like Sarah's proposed clarifications.

I also like Natalia's proposal at the top of this thread. The 3 (2?) month onboarding sounds good -- I believe I took about 2 or 3 months before I started firing off replies on "my own authority" just to manage the volume.

shaib replied:
> Does that make sense to you? Do you think any of those bits should be spelled out?

(it's a little hard to discuss this in a comment thread, when it's effectively "a PR on the PR", but I'll try)

> Also, any DSF member or member of a DSF working group may submit a nomination

I'd use "Additionally" instead of "Also".

> New members complete a 3-month onboarding

"New members, including new Fellows, complete..."

The point about Fellows leaving does not need explication IMO.

shaib replied:
>  I see three main scenarios for removal:
> 
>     * Inactivity: If a member is inactive per this document’s definition, the team should be able to remove them independently, with a note of thanks and an open invitation to return. Steering Council involvement doesn’t seem necessary here, nor does unanimous agreement from the team.
> 

Nor a vote, really. With inactivity, I see two scenarios:
- The member is inactive enough to fail to reaffirm their membership, and is removed for that
- The member, while otherwise inactive, does reaffirm their membership. In that case, a friendly removal with an open invitation to return is really just another request for reaffirmation, and is likely to achieve nothing.

>     * Security or malicious concerns: The team should be able to temporarily suspend access if there is a credible concern, notifying the individual and pausing access during investigation. A final decision (e.g., permanent removal) should involve the Steering Council. Temporary action should be quick. I would propose this requires at least two team members voicing a concern.
> 

Thanks for bringing that up -- I had completely failed to account for this situation. And I completely agree.

>     * Code of Conduct or behavioral issues: These cases are often complex and an ugly/messy business, so I am comfortable with the Steering Council (and/or CoC team) being mediators who handle the investigation and make the final decision (perhaps with a vote).
> 

Yes. Except -- on second thought, and taking @LilyFirefly 's note -- in all cases where oversight seems required (inactive-but-reaffirming members, malice, or other issues) -- the approving committee should probably be the Board, not the Steering Council.

shaib replied:
This thread requires more work as well

sarahboyce replied:
This would be my suggestion based off the discussion so far:
```suggestion
Any DSF member may submit a nomination, including a self-nomination, to the team's mailing list (security@djangoproject.com).
If the team needs to make a call for volunteers, it will be posted on the [djangoproject.com blog](https://www.djangoproject.com/weblog/) and the [Django Forum](https://forum.djangoproject.com). 

Nominations are approved/rejected during the team meeting.
Team members are encouraged to share their position via the mailing list before the meeting.
A nomination is approved when at least one existing Security Team member votes in favor and no member objects.
Note that there is no upper limit to the team size

New members, including new Fellows, complete a 3-month onboarding period during which all external communications (with reporters, vendors, or distributors) must be reviewed by an existing member before sending.
This ensures each report is handled consistently and no valid reports are inadvertently dismissed.

As Django Fellows are contracted to deal with security reports, all Django Fellows are automatically added to the Security Team.

## Membership management

Each year, every non-Fellow member will need to reaffirm their membership with the team
A non-Fellow member can leave for any reason at any time

### Reasons members can be removed from the team

#### Inactivity

If a member is inactive per this document’s definition, they should be removed from the team with a note of thanks and an open invitation to return.

#### Security or malicious concerns

If there is a credible security or malicious concern, a team member can report this in a private email to all other members.
If at least 2 members (including the reporter) share this concern, the team must temporarily suspend access during an investigation, notifying the individual.
A final decision (e.g., permanent removal) requires a vote from the Steering Council.

#### Code of Conduct working group disqualification

As an official Django space, the security team abides by the Code of Conduct.
As per the [Code of Conduct Enforcement Ladder](https://github.com/django/code-of-conduct/blob/main/enforcement-ladder.md), members may be temporarily suspended or permanently banned from the security team following a Code of Conduct report.
```

LilyFirefly replied:
> A final decision (e.g., permanent removal) requires a vote from the Steering Council.

Perhaps this should be the Board instead?

sarahboyce replied:
I am getting the feeling that the Steering Council wouldn't feel comfortable with the decision to ban a member due to security/malicious concerns. I feel like the Board would feel even less comfortable as they are likely to be less familiar with the inner workings of Django.
@LilyFirefly Can you say why you think the Steering Council is not appropriate?

Another option is to hold a vote within the security team.
Perhaps the temporary ban is for 3 months to conclude the investigation and form agreement with the team. An optional 1 month extension to reach a consensus. The ban becomes permanent if a majority decision is reached (at least 50% in favor of banning).
We could make the agreement type stronger (such as unanimous) but if half the team thinks this member is malicious/a security risk, I think it does not make sense for them to be in the team.

> As noted above, I disagree with the idea that the team should be able to kick out members without Steering Council approval.

@shaib I realise you stated this above but I think you weren't considering security/malicious concerns at the time

LilyFirefly replied:
My initial feeling was that removing people from teams is more on the people management side of things, which feels more like the Board than the Steering Council to me. But I see the point about maybe the Board doesn't have the technical knowledge that's needed here. That said, security awareness isn't a strong requirement to be on the Steering Council either, so Steering Council members might also not have the right technical knowledge either. Maybe this needs to be a joint discussion between the Security Team, the Board and the Steering Council?

shaib replied:
@sarahboyce No, I still feel the team shouldn't be able to kick out members without approval from an overseeing body. I agree that where there are suspicions of malice, a suspension needs to be as quick as possible, so I agree with your wording about suspension -- but a final decision to kick someone out should, IMO, require oversight.

I also noted [above](#discussion_r3142579152) that, like @LilyFirefly , I prefer the overseeing body for this to be the Board, rather than the SC. The reasoning is a stronger version of what LiLy said: The SC (and the Security Team) is a technical authority, but handing out sanctions should be handled by an organizational authority -- and that is the Board (and/or the CoC committee).

Just to make my position even clearer -- I don't think a deep understanding the inner workings of Django should be relevant for this sort of accusation. If technical arguments become material, I expect the Board to be able to judge them. Further, when something like this is done, there are decisions to be made about whether it is publicized, and there may be legal ramifications -- I can't see it being done properly without deep Board involvement.

sarahboyce replied:
OK then are we happy with the suggestion above replacing Steering Council with the DSF board?

shaib replied:
I am. Thanks!

sarahboyce replied:
Resolved by 9c1f116769bae4746ed4b14e5fad49bb1e5b6c73

nessita replied:
Sorry to chime in a bit late but given the very recent commentaries, when we think about "Security or malicious concerns" I do think this is a blunt CoC violation so in all cases other than inactivity I think the right body may be the CoC (potentially seeking support from other bodies).

@dryan What's your view?

tim-schilling replied:
> My initial feeling was that removing people from teams is more on the people management side of things, which feels more like the Board than the Steering Council to me.

@LilyFirefly I think the SC should be managing the technical teams where it can instead of the Board. And in fact we have been doing that a bit more with the CoC team.

tim-schilling replied:
If a person on the security team were to do something technical that would cause them to be removed from the team permanently, there'd be two parts to it. A technical review to confirm said action and then a code of conduct review to confirm it was against our CoC.

That technical review could be led by the chair of the security team (or another member). If the team doesn't want to lead it, then I would argue the Steering Council is the next most capable on paper since it's elected to shepherd the technical direction of the framework. I think board members would be capable as well, but they aren't expected to be involved in the development of the codebase.

The CoC evaluation should be done by the CoC WG and will almost certainly include work with the Board. Since members on the Security Team are extremely trusted by the community, they'll likely have access or involvements across the community. If they action is enough to warrant their removal from the team, they'll likely have other access removed as well which may require coordination from the Board. And since these members are trusted, these types of actions are usually reviewed by the Board any way. It won't happen in isolation.

In short, process-wise, this should be Security Team -> Steering Council -> CoC WG. Though actual implementation will involve coordination across several groups and won't be siloed to specific groups.

shaib replied:
If a person on the security team were to do something technical that would cause them to be removed from the team permanently, that would be corruption. I've reviewed the current Code of Conduct, and I see how that can be construed to fit under the general clause "Behaving in other ways that threaten the well-being of our community", but it is very different from the interpersonal character of all the specifically listed restricted behaviors.

I strongly disagree with this widening of the scope of the CoC WG's purview.

dryan replied:
I agree with this as well. The CoCWG should avoid action based purely on a technical issue. If I'm understanding the hypothetical here it would be a person putting malicious code into Django. That feels like it should be handled by the Security and Steering Council groups.

dryan replied:
Just to flesh this out a bit more, I think the security team (and all working groups) should be largely self-governing while coordinating with other groups when appropriate. For the security team here's how I would handle it:

* Removing an inactive team member is an internal vote: updating the relevant membership lists is likely notice enough to the broader community.
* Removing an unruly team member: if there's any question if the behavior violates the CoC then the behavior should get reported to the CoCWG. The CoCWG would then make a decision on the violation and recommend action that would include security team membership.
* Removing a malicious team member: that feels like a Steering Council involved-issue to me. Though I wouldn't be surprised if someone putting in malicious code also violated the CoC so both groups should get involved.

tim-schilling replied:
I think the current language of the charter generally fits that framework. The main difference being around the malicious team member being removed by a decision of the board instead of the steering council.

jefftriplett replied:
+1 to @dryan's breakdown. That framing (separating inactivity, security concerns, and Code of Conduct issues into distinct removal paths) is really helpful, and I agree with it.

Speaking with my board hat on: I'd prefer the working group have the ability to remove a member directly. Team self-governance for inactivity and fit is healthy, and it keeps the board out of routine team management.

For context on the broader authority question: the board already has the ability to remove any member from any working group as a matter of DSF governance. That backstop exists regardless of what this charter says. Given that, I'd lean toward the charter explicitly granting the WG its own removal power and not trying to fully enumerate the board's role here, so the document reads as team-empowering rather than board-looming. We can note the board retains its standing authority in a single line if people feel it needs to be said.

shaib replied:
Thanks @dryan , I find your comments very helpful.
@jefftriplett , I still see it as highly unlikely that, in a case of a malicious team member, anything permanent could be done without deep Board involvement. That said, I don't think it matters enough to block the charter.
FWIW, if we're explaining removals, and the board has the power to remove, I would like to have that one-line-mention added.

tim-schilling replied:
What about the following change @shaib @nessita? It suggests the team should try to handle this internally first, but can escalate things when needed.

```diff
- A final decision (e.g., permanent removal) requires a vote from the DSF Board.
+ If the team can't arrive at a consensus, the decision can be escalated to the DSF Board.
```

nessita replied:
Sounds good to me! Thanks for iterating.

shaib replied:
Agreed, it's fine. Thanks.

jefftriplett replied:
Works for me too. I think the updated wording is much less looming. Good change!

nessita commented on active/security.md:
```diff
@@ -0,0 +1,113 @@
+The team has discussions in the following places:
+
+1. Formal and sensitive discussions on the mailing list: security@djangoproject.com
+2. Informal and team discussions on the DSF Slack in the private channel `#security-team`
```
The Slack channel is named `security-private` (minor FYI).

RealOrangeOne replied:
Much like there's a public channel for the steering council, should there be a public one for the security team too? That way there's a single entrypoint for security-related conversations that don't necessarily need to be in the private team-only channel?

(Perhaps bikeshedding a little here).

jacobtylerwalls replied:
Let's leave that out for now: every new channel causes more fragmentation.

nessita reviewed (commented):
Did a whole pass and I focused on the unresolved comments. Thank you Tim! 🌟

tim-schilling committed:
Add chair

MarkusH commented on active/security.md:
```diff
@@ -0,0 +1,113 @@
+- The team/WG will vote (50%+1) to approve/deny new member.
+- The team will directly vote on new Chair/Co-Chairs
+- All Django Fellows are automatically added to the Security team
+- A Django Fellow contract termination removes the person from the Security team
```
Can we include something that states, that they're free to remain part of the team as any other member?

tim-schilling replied:
I think the reason we left that off from the Ops charter was that the member could be added as community member anyway and we didn't need to legislate it. I still think that's the better way to go, but I hold that opinion less strongly now :grin:

tim-schilling replied:
```suggestion
- A Django Fellow contract termination removes the person from the Security team, unless they prefer to remain on the team as a community member
```

jefftriplett replied:
Would this spell out? I'm not quite set on the second sentence, but it spells out their possible options: 

> A Django Fellow contract termination does not automatically determine continued Security team membership. A former Fellow may remain on the team as a community member, choose to step down, or have their continued membership considered separately.

shaib replied:
I think it's good to have a default, and I think it's good that the default is removal, so that staying needs to be explicit.

sarahboyce replied:
I believe this discussion became redundant by the rewrite of this section within 9c1f116769bae4746ed4b14e5fad49bb1e5b6c73 - please re-review and shout if you feel anything should be added in

shaib replied:
I read that rewrite as implicitly saying that a fellow stays on the team after contract termination -- because it doesn't list contract termination as a reason for removal. I am OK with that (my comment from a month ago notwithstanding), but I think it should still be explicit.

tim-schilling replied:
See suggestion here: https://github.com/django/dsf-working-groups/pull/56/changes#r3416804974

shaib commented on active/security.md:
```diff
@@ -0,0 +1,93 @@
+
+Members must opt-in to remain on the team on an annual basis. They may also leave for any reason.
+
+Members can also be removed by:
```
I think this discussion has been superseded elsewhere and can be resolved.

jacobtylerwalls commented on active/security.md:
```diff
@@ -0,0 +1,113 @@
+## Membership
+
+- Chair: Natalia Bidart
+- Co-Chair:
```
I can volunteer to be co-chair. I guess we can vote on this at the May meeting.

sarahboyce replied:
Resolved by 9c1f116769bae4746ed4b14e5fad49bb1e5b6c73

shaib commented on active/security.md:
```diff
@@ -0,0 +1,113 @@
+- Applying and backporting those fixes
+- Communicating with reporters
+- Communicating with the public about security releases
+- Communicating with operating-system vendors and other distributors of Django
```
We don't treat "distributors" as a category -- we have a prenotification list, which includes major distributors but also other important users.
```suggestion
- Maintaining a list of organizations to receive pre-notifications about security releases, and making these pre-notifications
```

sarahboyce replied:
Resolved by 9c1f116769bae4746ed4b14e5fad49bb1e5b6c73

shaib commented on active/security.md:
```diff
@@ -0,0 +1,113 @@
+
+There are specific roles that have a higher level of expectations:
+
+- **Chair / Co-Chair:** Responsible for coordinating the group, scheduling meetings, renewing the group’s membership, and ensuring that the group’s activities align with its scope and responsibilities. The Chair and Co-Chair roles should be re-evaluated annually by the team.
```
I think this may be a little clearer:
```suggestion
- **Chair / Co-Chair:** Responsible for coordinating the group, scheduling meetings, renewing the group’s membership, and ensuring that the group’s activities align with its scope and responsibilities. The Chair and Co-Chair should be re-elected annually by the team. The details of the election procedure are left to the team.
```

sarahboyce replied:
Resolved by 9c1f116769bae4746ed4b14e5fad49bb1e5b6c73

shaib commented on active/security.md:
```diff
@@ -0,0 +1,113 @@
+1. Formal and sensitive discussions on the mailing list: security@djangoproject.com
+2. Informal and team discussions on the DSF Slack in the private channel `#security-team`
+3. Monthly video-conference meetings
+
```
```suggestion
4. A private Github repository where patches are prepared
```
(is there harm in linking publicly to the private repo?)

sarahboyce replied:
I think we don't need to link as it doesn't add much value for the standard reader (as they wouldn't have access)

sarahboyce replied:
Resolved by 9c1f116769bae4746ed4b14e5fad49bb1e5b6c73

shaib commented on active/security.md:
```diff
@@ -0,0 +1,113 @@
+
+The team has two responsibilities in regards to reporting to the Board and the Steering Council:
+
+1. Use [Django Release Announcements thread](https://forum.djangoproject.com/t/django-release-announcements/655/96) on the Forum to report security releases
```
On one hand -- it seems odd to mention the Forum and not the [blog](https://www.djangoproject.com/weblog/); on the other hand -- is any of the two considered "reporting to the Board and SC", as opposed to "reporting to the public"?

sarahboyce replied:
I think point 1 should be removed as this is handled and documented as part of the process of a security release.

shaib replied:
That would be consistent enough for my taste :smile:

sarahboyce replied:
Resolved by 9c1f116769bae4746ed4b14e5fad49bb1e5b6c73

shaib reviewed (commented):
While we're getting close here, there are still a couple of important issues we need to resolve. In particular, 
- the issue of member removal doesn't seem to be consolidated, and
- the issue of responsibility/ownership of reports appears to be in conflict between what we might like to have and what we currently do

Other issues mentioned here are, I think, minor, or just have agreed comments that need to be applied.

RealOrangeOne commented on active/security.md:
```diff
@@ -0,0 +1,113 @@
+
+These team members are responsible for acknowledging and triaging reports initially to determine likelihood of security concern and severity. As this is a volunteer role, the Fellows will support the triagers and when necessary, handle the initial triaging.
+
+Every member can adopt and step back as a Report Triager as their schedule allows.
```
Thought: Should there be some kind of expected tenure for the Report Triager role? My thinking being that if people are free to rotate "as schedule allows", it may not be explicit when they do, so the entire team could step back assuming the rest of the team can pick things up.

sarahboyce replied:
Although I like the idea, I don't think we need to define a process here as Fellows would fill this role if we ever had a whole team step back

RealOrangeOne commented on active/security.md:
```diff
@@ -0,0 +1,113 @@
+- The team will directly vote on new Chair/Co-Chairs
+- All Django Fellows are automatically added to the Security team
+- A Django Fellow contract termination removes the person from the Security team
+- Each year, every non-Fellow member will need to reaffirm their membership with the team
```
Thought: Should this coincide with a specific date, or is it based on when they joined? It may be beneficial to have a single time this is discussed for all members, to also enable a single time when the team may be actively looking to recruit.

sarahboyce replied:
I agree this should be a specific date, such as in the last team meeting of the year

shaib replied:
+1

tim-schilling replied:
The specific date could be extracted into an internal document for the team to use. The Steering Council has a Google doc guide of sorts that highlights the various tools and processes in it. I would expect each team to have something similar. That can be updated as needed internally to fit whatever needs the team has without going through a public PR.

RealOrangeOne commented on active/security.md:
```diff
@@ -0,0 +1,113 @@
+- Report triagers: 
+- Steering Council Liaison (must be an active Steering Council member; may be the same as Chair/Co-Chair): Frank Wiles
+- Other members:
+  - Adam Johnson
+  - Jacob Walls
+  - Jake Howard
+  - Mariusz Felisiak
+  - Markus Holtermann
+  - Michael Manfre
+  - Natalia Bidart
+  - Paul McMillan
+  - Sarah Boyce
+  - Shai Berger
+  - Simon Charette
```
Thought: As part of formalizing the charter, should we pre-seed the triagers role with those interested? Currently it looks like the team is incredibly understaffed despite that (probably) not being the case.

sarahboyce replied:
I agree we should fill this but I don't think it's a requirement for merging (as we need to keep the list up-to-date anyway)

sarahboyce committed:
Update based off feedback from recent discussions.

sarahboyce reviewed (approved):
Thank you all!
Approving as I believe this could be merged as is, I will continue to engage in discussions as and when they pop up 👍

shaib commented on active/security.md:
```diff
@@ -0,0 +1,131 @@
+1. Formal and sensitive discussions on the mailing list: security@djangoproject.com
+2. Informal and team discussions on the DSF Slack in the private channel `#security-private`
+3. Monthly video-conference meetings
+4. A private Github repository where patches are prepared
```
Sorry to start yet another discussion on this line, but in the last team meetings we agreed to use issues on the private GH repo to collect and triage all incoming reports.

tim-schilling replied:
```suggestion
4. A private Github repository to collect and triage all reports and to prepare patches
```

Would this cover that?

shaib reviewed (commented):
There is one needed update and one possible update, because in team meetings, we agreed to use Issues in our private GH repo to record and triage all incoming reports.

The needed one is in the Comms section (see comment). The possible one is with regard to the last truly-open issue in the charter, that of the responsibility of a Triager: we now, at least in theory, do have the means to formally record and track ownership of issues, and so can make the Triager responsibilities as firm as originally intended. I think we can also merge the PR with the Triager role description as it stands, and wait to see if GH issues actually work well for us.

tim-schilling commented on active/security.md:
```diff
@@ -0,0 +1,131 @@
+## Membership management
+
+Each year, every non-Fellow member will need to reaffirm their membership with the team
+A non-Fellow member can leave for any reason at any time
```
```suggestion
Each year, every non-Fellow member will need to reaffirm their membership with the team.
A non-Fellow member can leave for any reason at any time. A Fellow whose contract ends is eligible to stay on the team if they would like.
```

RealOrangeOne commented on active/security.md:
```diff
@@ -0,0 +1,131 @@
+There are specific roles that have a higher level of expectations:
+
+- **Chair / Co-Chair:** Responsible for coordinating the group, scheduling meetings, renewing the group’s membership, and ensuring that the group’s activities align with its scope and responsibilities. The Chair and Co-Chair roles should be re-elected annually by the team. The details of the election procedure are left to the team.
+- **Report Triagers:** Acknowledge and triage initial reports and communicate with reporters.
```
Suggestion: My re-reading of this highlighted that it's not super clear the difference between triagers and "other members". Might be worth noting here?

```suggestion
- **Report Triagers:** Acknowledge and triage initial reports and communicate with reporters.
- **Other Members:** Assist the Report Triagers when needed, for example when a report matches their area of expertise.
```

tim-schilling committed:
Add mention of other members responsibilities along side triagers.

Co-authored-by: Jake Howard <6527489+RealOrangeOne@users.noreply.github.com>

tim-schilling committed:
Clarify purpose of the private github repo

tim-schilling committed:
Explicitly acknowledge a Fellow can stay on the team after their contract ends.

tim-schilling committed:
Update removing user flow to not require board involvement.
